docs/how-to/create-new-event.rst

```diff
@@ -0,0 +1,167 @@
+Therefore, we've put together a guide on how to add a new event to have an effective contribution process.
+
+
+1. Propose the new event to the community
+-----------------------------------------
+
+When creating a new event, you must justify its implementation. For example, you could create a post in Discuss,
+send a message through slack or open a new issue in the library repository listing your use cases for it. Or even,
+if you have time, you could accompany your proposal with the implementation of the event to illustrate its behavior.
+
+2. Place your event in an architecture subdomain
+-------------------------------------------------
+
+As specified in the Architectural Decisions Record (ADR) events naming and versioning, the event definition needs an Open edX Architecture
+Subdomain for:
+
+- The name of the event: ``{Reverse DNS}.{Architecture Subdomain}.{Subject}.{Action}.{Major Version}``
+- The package name where the definition will live, eg. ``learning/`` or ``content_authoring/``.
+
+For those reasons, after studying your new event purpose, you must place it in one of the subdomains supported by the project:
+
++----------------+----------------------------------------------------------------------------------------------------+
+| Subdomain name | Description                                                                                        |
++================+====================================================================================================+
+| Authoring      | Allows educators to create, modify, discover, package, annotate (tag), and share learning content. |
++----------------+----------------------------------------------------------------------------------------------------+
+| Learning       | Allows learners to consume content and perform actions in a learning activity on the platform.     |
++----------------+----------------------------------------------------------------------------------------------------+
+| Discovery      | Allows learners to find the right content at the right time to help achieve their learning goals.  |
++----------------+----------------------------------------------------------------------------------------------------+
+| Programs       | Allows educators and learners to manage and engage in bundled packages (programs) of learning.     |
++----------------+----------------------------------------------------------------------------------------------------+
+
+Refer to `edX DDD Bounded Contexts <https://openedx.atlassian.net/l/cp/vf8XjRiX>`_ confluence page for more documentation on domain-driven design in the Open edX project.
+
+3. Create the data attributes for the event (OEP-49)
+----------------------------------------------------
+
+Events send `data attributes <https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0049-django-app-patterns.html#data-py>`_ when triggered. Therefore, when designing your new event definition you must
```

Comment:
Somewhere in here I think we may want to talk about the difference between state-transfer events and notification events, and how the data attributes differ between the two. I found this to be a pretty useful guide:  https://medium.com/swlh/event-notification-vs-event-carried-state-transfer-2e4fdf8f6662

Reply:
You're right! Thank you for the suggestion! I'll try to pick up this work during the week and take a look at those resources

Reply:
We currently use the same amount of data when notifying, and for state transfer, I believe the only data class with old / new state is CertificateData, but it's not used. What do you think would be worth mentioning? Should we add optional states to the existing data classes? 🤔

Reply:
I am marking this as resolved, because I'd love to land these docs sooner rather than later, and I don't think this is a blocker. If anyone wants to research this and iterate on the documentation, that is fine.